nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/impl/OperationChainCompiler.java

```diff
@@ -91,7 +84,7 @@ public boolean equals(Object obj) {
                 return false;
             }
             Connector other = (Connector) obj;
-            return hashcode == other.hashcode;
+            return typeof.equals(other.typeof) && typein.equals(other.typein);
```

Comment:
Risk of NPE on `typeof` or `typein`?

Reply:
No the hashCode computation already dereferences them so we can consider them always non-null.